data.md

```diff
@@ -337,23 +347,84 @@ One needs to unname the `ValTypes` first before calling the `#take` or `#drop` f
 Strings
 -------
 
+Wasm use a different character escape rule with K, so we need to define the `unescape` function ourselves.
+
+```k
+    syntax String ::= unescape(String)              [function]
+                    | unescape(String, Int, String) [function, klabel(unescapeAux)]
+ // -------------------------------------------------------------------------------
+    rule unescape(S) => unescape(S, 1, "")
+    rule unescape(S, IDX, SB) => unescape(S, IDX +Int 1, SB +String substrString(S, IDX, IDX +Int 1))
+      requires IDX <Int lengthString(S) -Int 1
+       andBool substrString(S, IDX, IDX +Int 1) =/=K "\\"
+    rule unescape(S, IDX, SB) => unescape(S, IDX +Int 3, SB +String chrChar(String2Base(substrString(S, IDX +Int 1, IDX +Int 3), 16)))
+      requires IDX <Int lengthString(S) -Int 3
+       andBool substrString(S, IDX, IDX +Int 1) ==K "\\"
+       andBool (findChar("0123456789abcdefABCDEF", substrString(S, IDX +Int 1, IDX +Int 2), 0) =/=Int -1 )
+    rule unescape(S, IDX, SB) => SB
+      requires IDX ==Int lengthString(S) -Int 1
+
+    rule unescape(S, IDX, SB) => unescape(S, IDX +Int 2, SB +String chrChar(String2Base("09", 16)))
+      requires IDX <Int lengthString(S) -Int 2
+       andBool substrString(S, IDX, IDX +Int 1) ==K "\\"
+       andBool substrString(S, IDX +Int 1, IDX +Int 2) ==K "t"
+    rule unescape(S, IDX, SB) => unescape(S, IDX +Int 2, SB +String chrChar(String2Base("0A", 16)))
+      requires IDX <Int lengthString(S) -Int 2
+       andBool substrString(S, IDX, IDX +Int 1) ==K "\\"
+       andBool substrString(S, IDX +Int 1, IDX +Int 2) ==K "n"
+    rule unescape(S, IDX, SB) => unescape(S, IDX +Int 2, SB +String chrChar(String2Base("0D", 16)))
+      requires IDX <Int lengthString(S) -Int 2
+       andBool substrString(S, IDX, IDX +Int 1) ==K "\\"
+       andBool substrString(S, IDX +Int 1, IDX +Int 2) ==K "r"
+    rule unescape(S, IDX, SB) => unescape(S, IDX +Int 2, SB +String chrChar(String2Base("22", 16)))
+      requires IDX <Int lengthString(S) -Int 2
+       andBool substrString(S, IDX, IDX +Int 1) ==K "\\"
+       andBool substrString(S, IDX +Int 1, IDX +Int 2) ==K "\""
+    rule unescape(S, IDX, SB) => unescape(S, IDX +Int 2, SB +String chrChar(String2Base("27", 16)))
+      requires IDX <Int lengthString(S) -Int 2
+       andBool substrString(S, IDX, IDX +Int 1) ==K "\\"
+       andBool substrString(S, IDX +Int 1, IDX +Int 2) ==K "'"
+    rule unescape(S, IDX, SB) => unescape(S, IDX +Int 2, SB +String chrChar(String2Base("5C", 16)))
+      requires IDX <Int lengthString(S) -Int 2
+       andBool substrString(S, IDX, IDX +Int 1) ==K "\\"
+       andBool substrString(S, IDX +Int 1, IDX +Int 2) ==K "\\"
+
```

Comment:
... and that longer byte sequences can be encoded as escaped "Unicode", with `\u{<hex>+}`.

Btw, I think this is not right. It's not necessarily 4 hex nums, it can any multiplicity (over 1 at least), see page 108 and definition of hexnum at page 106. Also, there's this side condition: "(if n < 0xD800 ∨ 0xE000 ≤ n < 0x110000)". I don't know if that is worth adding, may be overkill, but would be cool to have if it's easy to add.